x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/external/request/elastic/ElasticInferenceServiceSparseEmbeddingsRequestTests.java

```diff
@@ -32,23 +33,37 @@ public void testCreateHttpRequest() throws IOException {
         var url = "http://eis-gateway.com";
         var input = "input";
 
-        var request = createRequest(url, input);
+        var request = createRequest(url, input, License.OperationMode.BASIC);
         var httpRequest = request.createHttpRequest();
 
         assertThat(httpRequest.httpRequestBase(), instanceOf(HttpPost.class));
         var httpPost = (HttpPost) httpRequest.httpRequestBase();
 
         assertThat(httpPost.getLastHeader(HttpHeaders.CONTENT_TYPE).getValue(), is(XContentType.JSON.mediaType()));
+        assertThat(httpPost.getLastHeader(ElasticInferenceServiceRequest.IS_TRIAL_LICENSE_HEADER).getValue(), is(Boolean.FALSE.toString()));
         var requestMap = entityAsMap(httpPost.getEntity().getContent());
         assertThat(requestMap.size(), equalTo(1));
         assertThat(requestMap.get("input"), is(List.of(input)));
     }
 
+    public void testLicenseTrialStatePropagatedThroughHTTPHeaders() {
+        var url = "http://eis-gateway.com";
```

Comment:
```suggestion
        var url = "http://elastic-inference-service.com";
```
Just a reminder that we don't want the term "gateway" in public-facing code or settings. I see that the other tests also use that, would be good to clean up at some point to hide our implementation detail